app/client/widgets/AttachmentsWidget.ts

```diff
@@ -142,21 +161,69 @@ export class AttachmentsWidget extends NewAbstractWidget {
     });
   }
 
+  private _setUploadingState(row: DataRowModel, uploading: boolean): void {
+    const rowId = row.getRowId();
+    const timeouts = this._uploadingTimeouts;
+    const states = this._uploadingStatesObs.get();
+    if (timeouts[rowId]) {
+      clearTimeout(timeouts[rowId]);
+    }
+    if (uploading) {
+      timeouts[rowId] = window.setTimeout(() => {
+        states[rowId] = true;
+        this._uploadingStatesObs.set({...states});
+        // @TODO: refactor this? this is done so that the row height is updated to have room for the loading spinner
+        const rowModel = this.field.viewSection().viewInstance()?.viewData.getRowModel(rowId);
+        if (rowModel) {
+          this.field.viewSection().viewInstance()?.viewData.trigger('rowModelNotify', [rowModel]);
+        }
```

Comment:
this is how I manage to make sure the row height expands if needed, in order to make sure the loading spinner always gets shown. I find it rather weird to have to go through all this stuff, so if you have any idea to refactor I'm all ears :)